### PR TITLE
finished extracting validation functions out of router files. Tweaked…

### PR DESCRIPTION
**Pull request by skwong25**
Finished extracting validation functions out of router files. 
Tweaked code to maximise the number of validation functions that can apply to both cat & breed routers.

### CODE REVIEW

File: src/breedsRouter.js [hunk 1]
```diff
@@ -1,70 +1,53 @@
...
+//  return next(err); 
+// 404 errors take an unfound id param
+// 400 errors take an error message param 
 
 const express = require('express');  
 const breedsRouter = express.Router();
 
 // imports class instance 
 const breedRepository = require('./breedsRepo'); 
+const validate = require('./validationFunctions');
```
Comment (fractalwrench) [inline src/breedsRouter.js:15]: nice consolidation 👌

File: src/validationFunctions.js [hunk 2]
```diff
@@ -1,58 +1,72 @@
 
-    // SIMPLE JS FUNCTIONS: 
-    function isInvalidString (value) {
-        return typeof value !== "string"                                  
-    }
+// SIMPLE JS FUNCTIONS: 
+function isInvalidString(value) {
```
Comment (fractalwrench) [inline src/validationFunctions.js:3]: `isInvalidString/isInvalidNum` are really good candidates for unit tests, so it'd be good to add a couple now that they're extracted into their own file.

File: src/validationFunctions.js [hunk 3]
```diff
@@ -1,58 +1,72 @@
...
-            if (invalidParam === true) {  
-                isError.push(`Error: Invalid ${key} parameter: "${value}"`)
-            }  
-        })      
-        let message = isError[0] || 'Check successful - object property values valid';
-        return message; 
-    }
+// checks for valid keys 
+function checkObjKeys(keys) {
+    const validKeys = ["name", "ageInYears", "favouriteToy", "description", "breedId"];
```
Comment (fractalwrench) [inline src/validationFunctions.js:37]: If possible I think it'd be preferable to extract `validKeys` into a parameter for this function. This has the advantage that it'd allow you to remove `checkBreedObjKeys`, and it also means that `validationFunctions.js` is better encapsulated and doesn't require specific knowledge about how cat/breed objects are validated.

**Review by fractalwrench — APPROVED** [review]
LGTM, pending a couple of minor comments. Thanks for the small PR, this was much more manageable to review 🎉

File: src/validationFunctions.js (hunk 3 above)
Comment (skwong25) [reply src/validationFunctions.js:37]: ooh good idea!

**Pull request merged**